src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -13484,7 +13484,8 @@ bool Unit::UpdatePosition(float x, float y, float z, float orientation, bool tel
         return false;
     }
 
-    bool const turn = (GetOrientation() != orientation);
+    // Check if angular distance changed
+    bool const turn = M_PI - fabs(fabs(GetOrientation() - orientation) - M_PI) > 0.001f;
```

Comment:
Position::NormalizeOrientation(GetOrientation() - orientation) // result is now in [0, 2PI) range
then turn could be expressed as `turn = G3D::fuzzyGt(Position::NormalizeOrientation(GetOrientation() - orientation), 0.0f);`